<launch>

<include file="cube_display.launch"/>

<node pkg="sr_slam" type="gyro_show" name="sr_gyro_show" output="screen">
  <!-- display style -->
  <param name="config/cloud_display_type"             value="TRIANGLES"  /> <!-- POINTS -->

  <param name="config/load_imu_data"                 value="true"/> <!-- whether to use imu data -->
  <param name="config/imu_file_dir"                  value="/media/work/work/data/sr4000/imu_fuse/imu_creep_ob1.dat"/> <!-- where to load imu file -->
  <param name="config/camera_syn_data"               value="/media/work/work/data/sr4000/imu_fuse/ftiming_creep_ob1.dat" /> <!-- timestamp for camera data -->
  <param name="config/b_test_imu_vro"                value="false" /> <!-- true whether to compare imu and vro -->

  <param name="imu_camera_syn_id"             value="50"/> <!-- this value used to synchronize the imu data and camera data-->

  <param name="sr_data_file_dir"              value="/media/work/work/data/sr4000/imu_fuse/frames_creep_ob1" />
  <param name="sr_data_prefix"                value="d1"/>
  <param name="sr_data_suffix"                value="bdat"/> <!-- dat -->
  
  <!-- syn_start_frame: dataset_3: 12 - 1100, dataset_2: 2 - 2310, dataset_1: 3-1200 -->
  <param name="sr_start_frame"                value="1"/> <!-- notice: for matching timestamp in ground_truth, start should be 1 -->
  <param name="sr_end_frame"                  value="7360"/> 
  <param name="sr_new_file_version"           value="false"/> <!-- true -->
  
  <param name="config/process_node_method"            value="display" /> <!-- slam, write, display  -->
  <param name="b_downsample_pc"               value="true"/> <!-- weather to downsample the point cloud -->
  <param name="filter_voxel_size"             value="0.01"/> <!-- voxel grid size -->

  <!-- display the 3D map, and reproject 2D point -->
  <param name="config/store_pointclouds"      value="true"/> 
  <param name="config/visualize_keyframes_only" value="true"/>
  <param name="config/glwidget_without_clouds" value="false"/>
 
  <!-- Visual Features, to activate GPU-based features see CMakeLists.txt -->
  <param name="config/feature_detector_type"         value="SIFT"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/feature_extractor_type"        value="SIFT"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/min_keypoints"                 value="50"/><!-- Extract no less than this many ... -->
  <param name="config/min_matches"                   value="10"/> <!-- Minimal number of matched features before RANSAC 29? -->
  <param name="config/nn_distance_ratio"             value="0.5"/> <!-- 0.5 for SIFT, 0.95 for SIFTGPU Feature correspondence is valid if distance to nearest neighbour is smaller than this parameter times the distance to the 2nd neighbour. This needs to be 0.9-1.0 for SIFTGPU w/ FLANN, since SIFTGPU Features are normalized-->

  <param name="config/max_dist_for_inliers"         value="0.05" /> <!-- 6 3 "Mahalanobis distance for matches to be considered inliers by ransac"-->
 
  <!-- vro strategy -->
  <param name="config/vro_strategy"                 value="vro_my" /> <!-- " VRO strategy: VRO_ori, VRO_my, VRO_plane, VRO_plane_em" -->
  <param name="config/ransac_iterations"            value="5000" /> <!-- "Number of iterations for registration" -->
  <param name="config/b_ransac_fixed_iter"           value="false"/> <!-- weather to use fixed   -->

  <!-- sift and siftgpu parameters -->
  <param name="sift_num_features"                   value="0" /> <!-- number of returned sift features -->
  <param name="sift_octave_layers"                  value="5" /> <!-- number octave layers -->
  <param name="sift_contrast_threshold"             value="0.04" /> <!-- initial threshold for keypint detection -->
  <param name="siftgpu_contrast_threshold"          value="0.0008" /> <!-- 0.0018 initial DOG threshold for siftgpu -->
  <param name="siftgpu_edge_threshold"              value="10." /> <!-- edge threshold for siftgpu  -->

  <!-- Algortithm settings -->
  <param name="config/optimizer_skip_step"    value="20" /><!-- 20 Optimize every n-th frame. Set negative for offline operation -->
  <param name="config/min_translation_meter"         value="0.05"/><!-- 0.05 frames with motion less than this, will be omitted -->
  <param name="config/min_rotation_degree"           value="2"/><!-- frames with motion less than this, will be omitted -->
  <param name="config/predecessor_candidates"        value="10"/><!-- 5 search through this many immediate predecessor nodes for corrspondences -->
  <param name="config/neighbor_candidates"           value="0"/><!-- 5 search through this many graph neighbour nodes for corrspondences -->
  <param name="config/min_sampled_candidates"        value="0"/><!-- 5 search through this many uniformly sampled nodes for corrspondences -->
  <param name="config/max_connections"               value="5"/><!-- 10 -1 stop after this many succesfully found spation relations -->


</node>
</launch>
